gtsam/hybrid/tests/testDCSAM.cpp

```diff
@@ -0,0 +1,685 @@
+  }
+
+  BetweenFactor<Pose2> bw(x0, X(7), dx * noise, meas_noise);
+
+  dcsam.update(graph, initialGuess);
+  HybridValues dcvals = dcsam.calculateEstimate();
+
+  // Values from the DCSAM repo
+  Values expectedContinuous;
+  expectedContinuous.insert<Pose2>(
+      X(0), Pose2(-3.53708e-33, -8.97559e-34, 4.1955e-33));
+  expectedContinuous.insert<Pose2>(X(1), Pose2(1, 0.0141421, 0.795398));
+  expectedContinuous.insert<Pose2>(X(2), Pose2(1.6899, 0.738184, 1.5908));
+  expectedContinuous.insert<Pose2>(X(3), Pose2(1.65576, 1.7377, 2.38619));
+  expectedContinuous.insert<Pose2>(X(4), Pose2(0.918069, 2.41298, -3.10159));
+  expectedContinuous.insert<Pose2>(X(5), Pose2(-0.0805657, 2.35886, -2.30619));
+  expectedContinuous.insert<Pose2>(X(6), Pose2(-0.740961, 1.60781, -1.5108));
+  expectedContinuous.insert<Pose2>(X(7), Pose2(-0.66688, 0.61046, -0.715398));
+
+  EXPECT(assert_equal(expectedContinuous, dcvals.nonlinear(), 1e-5));
+}
+
+/**
+ * This is a basic qualitative octagonal pose graph SLAM test
+ * to verify that DCSAM works on standard SLAM examples
+ * in the *incremental* setting.
+ */
+TEST(DCSAM, SimpleSlamIncremental) {
+  using namespace slam_fixture;
+
+  // Make a factor graph
+  HybridNonlinearFactorGraph graph;
+
+  // Values for initial guess
+  Values initialGuess;
+
+  Key x0 = X(0);
+
+  PriorFactor<Pose2> p0(x0, pose0, prior_noise);
+
+  initialGuess.insert(x0, pose0);
+  graph.push_back(p0);
+
+  // Setup dcsam
+  DCSAM dcsam;
+  dcsam.update(graph, initialGuess);
+
+  graph.resize(0);
+  initialGuess.clear();
+
+  Pose2 odom(pose0);
+
+  for (size_t i = 0; i < 7; i++) {
+    Key xi = X(i);
+    Key xj = X(i + 1);
+
+    Pose2 meas = dx * noise;
+
+    BetweenFactor<Pose2> bw(xi, xj, dx * noise, meas_noise);
+    graph.push_back(bw);
+
+    odom = odom * meas;
+    initialGuess.insert(xj, odom);
+    dcsam.update(graph, initialGuess);
+
+    graph.resize(0);
+    initialGuess.clear();
+  }
+
+  BetweenFactor<Pose2> bw(x0, X(7), dx * noise, meas_noise);
+
+  dcsam.update(graph, initialGuess);
+  HybridValues dcvals = dcsam.calculateEstimate();
+
+  // Values from the DCSAM repo
+  Values expectedContinuous;
+  expectedContinuous.insert<Pose2>(
+      X(0), Pose2(-3.53708e-33, -8.97559e-34, 4.1955e-33));
+  expectedContinuous.insert<Pose2>(X(1), Pose2(1, 0.0141421, 0.795398));
+  expectedContinuous.insert<Pose2>(X(2), Pose2(1.6899, 0.738184, 1.5908));
+  expectedContinuous.insert<Pose2>(X(3), Pose2(1.65576, 1.7377, 2.38619));
+  expectedContinuous.insert<Pose2>(X(4), Pose2(0.918069, 2.41298, -3.10159));
+  expectedContinuous.insert<Pose2>(X(5), Pose2(-0.0805657, 2.35886, -2.30619));
+  expectedContinuous.insert<Pose2>(X(6), Pose2(-0.740961, 1.60781, -1.5108));
+  expectedContinuous.insert<Pose2>(X(7), Pose2(-0.66688, 0.61046, -0.715398));
+
+  EXPECT(assert_equal(expectedContinuous, dcvals.nonlinear(), 1e-5));
+}
+
+/**
+ * This test is a sanity check that DCSAM works correctly for solely
+ discrete problems
```

Comment:
Same as for the continuous case - maybe not super useful? But could be good to make sure nothing is fundamentally broken...